incubator/buzzfeed-sso/README.md

```diff
@@ -0,0 +1,183 @@
+
+Many of the Kubernetes OAuth solutions require to run an extra container within the pod using [oauth2_proxy](https://github.com/bitly/oauth2_proxy), but the project seems to not be maintained anymore. The approach presented on this chart allows to have a global OAuth2 Proxy that can protect services even in different namespaces, thanks to [Kube DNS](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/).
+
+We use this chart in production at [MindDoc](https://minddoc.de) for protecting endpoints that have no built-in authentication (or that would require to run inner containers), like `Kibana`, `Prometheus`, etc...
+
+## Introduction
+
+This chart creates a SSO deployment on a [Kubernetes](http://kubernetes.io)
+cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.8+ with Beta APIs enabled
+- Kube DNS
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```bash
+$ helm install --name my-release stable/buzzfeed-sso
+```
+
+The command deploys SSO on the Kubernetes cluster using the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+
+This chart has required variables, see [Configuration](#configuration).
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
+
+```bash
+$ helm delete --purge my-release
+```
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following table lists the configurable parameters of the SSO chart and their default/required values.
+
+Parameter | Description | Default
+--- | --- | ---
+`namespace` | namespace to use | `default`
+`emailDomain` | the sso email domain for authentication | REQUIRED
+`rootDomain` | the parent domain used for protecting your backends | REQUIRED
+`auth.domain` | the auth domain used for OAuth callbacks | REQUIRED
+`auth.replicaCount` | desired number of auth pods | `1`
+`auth.resources` | resource limits and requests for auth pods | `{ limits: { memory: "256Mi", cpu: "200m" }}`
+`auth.nodeSelector` | node selector logic for auth pods | `{}`
+`auth.tolerations` | resource tolerations for auth pods | `{}`
+`auth.affinity` | node affinity for auth pods | `{}`
+`auth.service.type` | type of auth service to create | `ClusterIP`
+`auth.service.port` | port for the http auth service | `80`
+`auth.secret` | secrets to be generated randomly with `openssl rand -base64 32 | head -c 32`. | REQUIRED if `auth.customSecret` is not set
+`auth.tls` | tls configuration for central sso auth ingress. | `{ secretName: "sso-auth-tls-secret" }`
+`auth.customSecret` | the secret key to reuse (avoids secret creation via helm) | REQUIRED if `auth.secret` is not set
+`proxy.virtualHost` | wildcard domain for redirecting SSO to the backends | REQUIRED
+`proxy.providerUrlInternal` | url for split dns deployments |
```

Comment:
@darioblanco I believe this should be better documented because I just can't understand how to use this :)

Reply:
This is a new feature from the latest sso update. TBH I haven't tried it and the setting is there, plus there wasn't much documentation about it in the buzzfeed sso project.

Do you mean that the chart does not work? If you don't provide the `proxy.providerUrlInternal` it is just not set in the environment variables. Can you please explain a bit what is the problem here?

Reply:
@darioblanco from my tests, without this setting I couldn't make the chart work: the proxy would refuse my requests even though I had been authenticated first by the auth service.
By setting `proxy.providerUrlInternal` to the internal name in the cluster of the auth service, it started to work, but maybe I made a mistake elsewhere also and thought that was the problem?

This is what I put in the template:
```
            - name: PROVIDER_URL_INTERNAL
              {{- if .Values.proxy.providerUrlInternal }}
              value: {{ .Values.proxy.providerUrlInternal | quote }}
              {{- else }}
              value: "http://{{ template "buzzfeed-sso.fullname" . }}-auth:80"
              {{- end }}
```